src/Controller/PastriesController.php

```diff
@@ -27,10 +147,244 @@ public function displayPage(Request $request, Response $response, $args)
     }
+
+    public function getList(Request $request, Response $response, $args)
+    {
+        // GET parameters
+        $params = $request->getQueryParams();
+
+        /** @var UserFrosting\Sprinkle\Account\Authorize\AuthorizationManager */
+        $authorizer = $this->ci->authorizer;
+
+        /** @var UserFrosting\Sprinkle\Account\Database\Models\User $currentUser */
+        $currentUser = $this->ci->currentUser;
+
+        // Access-controlled page
+        if (!$authorizer->checkAccess($currentUser, 'see_pastries')) {
+            throw new ForbiddenException();
+        }
+        /** @var UserFrosting\Sprinkle\Core\Util\ClassMapper $classMapper */
+        $classMapper = $this->ci->classMapper;
+
+        $sprunje = new PastrySprunje($classMapper, $params);
+
+        // Be careful how you consume this data - it has not been escaped and contains untrusted user-supplied content.
+        // For example, if you plan to insert it into an HTML DOM, you must escape it on the client side (or use client-side templating).
+        return $sprunje->toResponse($response);
+    }
+
+    public function getModalCreate(Request $request, Response $response, $args)
+    {
+        // GET parameters
+        $params = $request->getQueryParams();
+
+        /** @var \UserFrosting\Sprinkle\Account\Authorize\AuthorizationManager $authorizer */
+        $authorizer = $this->ci->authorizer;
+
+        /** @var \UserFrosting\Sprinkle\Account\Database\Models\Interfaces\UserInterface $currentUser */
+        $currentUser = $this->ci->currentUser;
+
+        // Access-controlled page
+        if (!$authorizer->checkAccess($currentUser, 'see_pastries')) {
+            throw new ForbiddenException();
+        }
+
+        /** @var \UserFrosting\I18n\MessageTranslator $translator */
+        $translator = $this->ci->translator;
+
+        // Load validation rules
+        $schema = new RequestSchema('schema://requests/pastry/create.yaml');
+        $validator = new JqueryValidationAdapter($schema, $translator);
+
+        // Create a dummy pastry to prepopulate fields
+        $pastry = new Pastries();
+
+        $pastry->icon = 'fa fa-cutlery';
+
+        $fields = [
+            'hidden'   => [],
+            'disabled' => []
+        ];
+
+        return $this->ci->view->render($response, 'modals/pastries.html.twig', [
+            'pastry' => $pastry,
+            'form'   => [
+                'action'      => 'api/pastries',
+                'method'      => 'POST',
+                'fields'      => $fields,
+                'submit_text' => 'Create'
+            ],
+            'page' => [
+                'validators' => $validator->rules('json', false)
+            ]
+        ]);
+    }
+
+    public function getModalDelete(Request $request, Response $response, $args)
+    {
+        // GET parameters
+        $params = $request->getQueryParams();
+
+        $pastry = Capsule::table('pastries')->where('name', '=', $params['name'])->first();
```

Comment:
The `Patries` model could be used here instead of `Capsule`. Unless you want to show an alternative way to access a table. And I think this works too (on the model at least) : `->where('name',  $params['name'])`. Can actually see it on Line 305

Reply:
Fixed in last commit.